src/coreclr/vm/ceemain.cpp

```diff
@@ -1723,10 +1725,27 @@ struct TlsDestructionMonitor
                     thread->m_pFrame = FRAME_TOP;
                     GCX_COOP_NO_DTOR_END();
                 }
+#ifdef _DEBUG
+                BOOL oldGCOnTransitionsOK = thread->m_GCOnTransitionsOK;
+                thread->m_GCOnTransitionsOK = FALSE;
+#endif
+                if (!IsAtProcessExit() && !g_fEEShutDown)
+                {
+                    GCX_COOP_NO_DTOR();
+                    FreeThreadStaticData(&t_ThreadStatics, thread);
+                    GCX_COOP_NO_DTOR_END();
+                }
+#ifdef _DEBUG
+                thread->m_GCOnTransitionsOK = oldGCOnTransitionsOK;
+#endif
                 thread->DetachThread(TRUE);
             }
+            else
```

Comment:
Can you flip the predicate here and make this the top part of the if check?

Reply:
Why? That seems less clear to me.

Reply:
I typically place the smaller block of code first. It is always annoying to scroll through pages of source and then see an else that is a single line. Just something I tend to do, but if you prefer it this way so be it.